File: src/test/programmers/python/level1/GetReportResults.py
def solution(id_list, report, k):
    dict_report = {name: [] for name in id_list}
    dict_count = {name: 0 for name in id_list}
    stop = []
    result = []

    for i in report:
        user, get_report_user = i.split()
        if get_report_user not in dict_report[user]:
            dict_report[user].append(get_report_user)

    for i in id_list:
        for j in dict_report:
            if i in dict_report[j]:
                dict_count[i] += 1
        if dict_count[i] >= k:
            stop.append(i)

    for i in dict_report:
        count = 0
        for j in dict_report[i]:
            if j in stop:
                count += 1
        result.append(count)

    return result

id_list = ["muzi", "frodo", "apeach", "neo"]
report = ["muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"]
k = 2

id_list2 = ["con", "ryan"]
report2 = ["ryan con", "ryan con", "ryan con", "ryan con"]
k2 = 3

print(solution(id_list2, report2, k2))

File: src/test/programmers/java/level1/GetReportResults.java
package test.programmers.java.level1;

import java.util.*;

public class GetReportResults {
    public static void main(String[] args) {
        String[] id_list = {"muzi", "frodo", "apeach", "neo"};
        String[] report = {"muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"};
        int k = 2;
        System.out.println(Arrays.toString(solution(id_list, report, 2)));
    }

    static public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        Map<String, Set<String>> userReport = new HashMap<>();
        Map<String, Integer> userCount = new HashMap<>();
        List<String> stop = new ArrayList<>();

        for(String rep : report) {
            String[] s = rep.split(" ");
            String key = s[0], value = s[1];
            Set<String> set = userReport.getOrDefault(key, new HashSet<>());
            set.add(value);
            userReport.put(key, set);
        }

        for(String id : id_list) {
            Set<String> users = userReport.get(id);
            if(users == null) {
                continue;
            }
            for(String user : users) {
                userCount.put(user, userCount.getOrDefault(user, 0) + 1);
            }
        }

        for(String id : id_list) {
            if(userCount.get(id) != null && userCount.get(id) >= k) {
                stop.add(id);
            }
        }

        for(int i = 0; i < id_list.length; i++) {
            int count = 0;
            String id = id_list[i];
            Set<String> users = userReport.get(id);

            if(users == null) {
                answer[i] = 0;
                continue;
            }

            for(String user : users) {
                if(stop.contains(user)) count++;
            }

            answer[i] = count;
        }

        return answer;
    }
}
